xapian-letor/tests/api_letor.cc

```diff
@@ -291,37 +321,34 @@ DEFINE_TESTCASE(preparetrainingfile, generated)
     string file1;
     string file2;
     while (getline(if1, file1)) {
-	TEST(getline(if2, file2));
-	istringstream iss1(file1);
-	istringstream iss2(file2);
-	string temp1;
-	string temp2;
-	int i = 0;
-	while ((iss1 >> temp1) && (iss2 >> temp2)) {
-	    if (i == 0 || i == 1 || i == 21) {
-		TEST_EQUAL(temp1, temp2);
-	    } else {
-		size_t t1 = temp1.find_first_of(':');
-		size_t t2 = temp2.find_first_of(':');
-		TEST(abs(stod(temp1.substr(t1 + 1)) -
-			 stod(temp2.substr(t2 + 1))) <
-		     0.001);
-	    }
-	    i++;
+    TEST(getline(if2, file2));
+    istringstream iss1(file1);
+    istringstream iss2(file2);
+    string temp1;
+    string temp2;
+    int i = 0;
+    while ((iss1 >> temp1) && (iss2 >> temp2)) {
+	if (i == 0 || i == 1 || i == 21) {
+	TEST_EQUAL(temp1, temp2);
+	} else {
+	size_t t1 = temp1.find_first_of(':');
+	size_t t2 = temp2.find_first_of(':');
+	TEST_EQUAL_DOUBLE(stod(temp1.substr(t1 + 1)),
+			  stod(temp2.substr(t2 + 1)));
 	}
-	TEST_REL(i, ==, 22);
-	TEST(!(iss2 >> temp2));
+	i++;
+    }
+    TEST_REL(i, ==, 22);
+    TEST(!(iss2 >> temp2));
     }
     TEST(!getline(if2, file2));
-
     return true;
 }
 
-
 DEFINE_TESTCASE(preparetrainingfilethree_missingQrel, generated)
 {
-    string db_path = get_database_path("apitest_listnet_ranker3	",
-				       db_index_three_documents);
+    string db_path = get_database_path("apitest_listnet_ranker3 ",
```

Comment:
This string is what the database gets called - it really shouldn't include **any** whitespace, otherwise we end up with database filenames with spaces in, and they're just unnecessarily harder to work with if you need to poke test databases by hand during development.